### PR TITLE
Optional config parameter for user-agent header, alias for OSMGeocoder

### PR DESCRIPTION
**Pull request by chriso0710**
Made the following changes regarding issue #194 :
- new optional parameter Geokit::Geocoders::useragent
- send user agent header for both net adapters
- some small readme changes

Sorry for not providing any tests, but I was having some issues with 

```
rake test
```

and as I am not familiar with vcr and simplecov, I was not very keen to debug the issue :-)

Additionally I created a new alias OsmGeocoder because of the following error in Multigeocoding with OSM:

```
E, [2016-10-27T17:37:51.610364 #9174] ERROR -- : Caught an error during Multi geocoding call: uninitialized constant Geokit::Geocoders::OsmGeocoder
E, [2016-10-27T17:37:51.610421 #9174] ERROR -- : /Users/co/projects/geokit/lib/geokit/multi_geocoder.rb:65:in `const_get'
/Users/co/projects/geokit/lib/geokit/multi_geocoder.rb:65:in `geocoder'
/Users/co/projects/geokit/lib/geokit/multi_geocoder.rb:32:in `block in do_geocode'
/Users/co/projects/geokit/lib/geokit/multi_geocoder.rb:31:in `each'
/Users/co/projects/geokit/lib/geokit/multi_geocoder.rb:31:in `do_geocode'
/Users/co/projects/geokit/lib/geokit/geocoders.rb:90:in `geocode'
```

### CODE REVIEW

**Comment by mnoack** [comment]
I mostly like the Pull Request, I think it just needs a test that verifies if Net::HTTP receives the user agent.

e.g. mock 'Net::HTTP::Get' class
and see if that mock recevies 'new' with the URI and headers containing the agent. And of course a test that if user agent isn't set that it receives empty headers.

**Comment by chriso0710** [comment]
Thanks, I will give it a try with webmock next week...

**Comment by chriso0710** [comment]
Now here is my shot at webmock for testing the netadapters useragent. Hope it's ok this way. Please feel free to modify it.

I also fixed a bug with missing options param in bing geocoder, when using MultiGeocoder.
Added the :provider_order param in reverse_geocode. Same as in geocode.

File: lib/geokit/geocoders/openstreetmap.rb [hunk 1]
```diff
@@ -110,4 +110,6 @@ def self.set_bounds(result_json, loc)
       end
     end
   end
+
+  Geokit::Geocoders::OsmGeocoder = Geokit::Geocoders::OSMGeocoder
```
Comment (mnoack) [inline lib/geokit/geocoders/openstreetmap.rb]: Why did you do this. I feel we're just adding another alias when people are already using OSM, if anything the class should reflect the file name (or file name reflect the class, e.g. OpenStreetMapGeocoder

File: lib/geokit/net_adapter/typhoeus.rb [hunk 2]
```diff
@@ -2,7 +2,8 @@ module Geokit
   module NetAdapter
     class Typhoeus
       def self.do_get(url)
-        ::Typhoeus.get(url.to_s)
+        Geokit::Geocoders.useragent ? headers = {'User-Agent' => Geokit::Geocoders.useragent} : headers = {}
```
Comment (mnoack) [inline lib/geokit/net_adapter/typhoeus.rb]: Can you address this comment, e.g. headers = Geokit::Geocoders.useragent ? {'User-Agent' => Geokit::Geocoders.useragent} : {}

**Comment by mnoack** [comment]
@chriso0710 I've fixed tests in https://github.com/geokit/geokit/pull/202 so am closing this. Thanks very much for contributing to the fix.

**Comment by chriso0710** [comment]
My pleasure, thanks for merging :-)

**Pull request closed without merging**